### DISCUSSION TITLE
✔️ Add level up

### DISCUSSION THREAD

**Discussion by kevingrillet** [zebscripts/AFK-Daily#35] [Ideas]
I don't know if it's possible, but, I think this path should work:

> Resonating Crystal > Red Square? > No, skip // Yes, Open: 
> - New slot? > if yes, add
> - Strenghten Crystal if nothing red in the line
> - Skip if level up %5

**Comment by Zebiano** [comment]
I'm not a big fan of this feature, as adding random heroes to the resonating crystal or automatically strengthening it is in my opinion, not the best idea. I'd rather have users decide what they want to do with their crystal. Of course, it would be possible to implement this and then have users decide in the `config.sh` if they want to use the feature or not, but I do not see any real benefit from it. This would be a really low priority feature in my eyes, but if more people are interested, I might consider putting it higher up.

**Comment by kevingrillet** [comment]
I don't think that clicking on Strenghen Crystal is a problem, all players need to do it every day anyway...
And by new slow a wanted to say unlock a new slot with the free points, not adding a random hero :D

![Bluestacks_2021-04-21_12-41_oR4PA](https://user-images.githubusercontent.com/7203617/115540975-e8d90000-a29e-11eb-8826-30fe70ae1d2c.png)

**Reply by Zebiano** [reply]
> I don't think that clicking on Strenghen Crystal is a problem, all players need to do it every day anyway...

I disagree. Various players purposefully do not go up in level because they want to stay on a certain "level floor" to gain an advantage. [Take a look at this.](https://www.reddit.com/r/afkarena/comments/hq78fl/how_do_the_level_brackets_work/) It does look like it's not as predominant as before, but I'd still rather leave players the choice to do so. Or better yet, let players decide through the `config.sh` if they want to level up the crystal or not!

> unlock a new slot with the free points

Ooh! Yes! That is totally possible to do. The game already asks if you want to unlock a new slot or not, so it's a matter of checking for a blue button to unlock it or not.

**Reply by kevingrillet** [reply]
> Take a look at this.

I have played this game since... day 1 or 2.

I did stick to level 299 for a bit, and did the [Mythic Gears Trick](https://afk.guide/mythic-gear-trick/), but if you want to go after world 30 it loose its meaning and you need to level up and try to push as hard as possible (or wait 2 days).

>  Or better yet, let players decide through the config.sh if they want to level up the crystal or not!

The `config.sh` will end with a boolean for every functionality :p

**Reply by Zebiano** [reply]
Yeah, you're right, that's what it's going to be in the end... 😄

**Reply by kevingrillet** [reply]
I will do a pull request tomorrow, but I think it works:

```shell
# If red square, strenghen Crystal
function strenghenCrystal() {
    input tap 750 1025 # Crystal
    sleep 3

    # TODO: Detect if free slot, and take it.

    input tap 550 1850 # Strenghen Crystal
    sleep 1
    input tap 200 1850 # If level up %5 = 0
    sleep 1

    input tap 70 1810

    wait
    verifyRGB 20 1775 d49a61 "Strenghened crystal." "Failed to strenghen crystal."
}
```

I'm quite far from the free slot, but I will update it.

And I did add all the booleans to my [`config.sh`](https://github.com/kevingrillet/AFK-Daily/blob/master/deploy.sh) but the readme is not up to date... (#27)

**Reply by Zebiano** [reply]
Thank you! I'll be waiting for the PR.

**Reply by kevingrillet** [reply]
Did try this morning and it's working properly \o/
